src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -313,9 +343,46 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
         return false;
       }
     }
-    LOG.info("It is ok to repair segment '{}' on repair run with id '{}'",
-        segment.getId(), segment.getRunId());
-    return true;
+    if(gotMetricsForAllHosts) {
+      LOG.info("It is ok to repair segment '{}' on repair run with id '{}'",
+          segment.getId(), segment.getRunId());
+    } 
+    else {
+      LOG.info("Not ok to repair segment '{}' on repair run with id '{}' because we couldn't get all hosts metrics :'(",
+          segment.getId(), segment.getRunId());
+    }
+    
+    return gotMetricsForAllHosts; // check if we should postpone when we cannot get all metrics, or just drop the lead
+  }
+  
+  private void handlePotentialStuckRepairs(JmxProxy hostProxy, LazyInitializer<Set<String>> busyHosts, String hostName) throws ConcurrentException {
+    if (!busyHosts.get().contains(hostName) && context.storage.getStorageType() != StorageType.CASSANDRA) {
+      LOG.warn("A host ({}) reported that it is involved in a repair, but there is no record "
+          + "of any ongoing repair involving the host. Sending command to abort all repairs "
+          + "on the host.", hostProxy.getHost());
+      hostProxy.cancelAllRepairs();
+    }
+  }
+  
+  private Optional<HostMetrics> getMetricsForHost(String hostName, JmxProxy hostProxy) {
+    try {
+      int pendingCompactions = hostProxy.getPendingCompactions();
+      boolean hasRepairRunning = hostProxy.isRepairRunning();
+      
+      HostMetrics metrics = HostMetrics.builder().withHostAddress(hostName)
+                                                 .withPendingCompactions(pendingCompactions)
+                                                 .withHasRepairRunning(hasRepairRunning)
+                                                 .withActiveAnticompactions(0) // for future use
+                                                 .build();
+      
+      context.storage.storeHostMetrics(metrics);
+      
+      return Optional.fromNullable(metrics);
+      
+    } catch(Exception e) {
+      LOG.info("Cannot reach node {} through JMX. Trying to get metrics from storage...", hostName, e);
+      return context.storage.getHostMetrics(hostName);
+    }
   }
 
   private boolean repairHasSegmentRunning(long repairRunId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "repairHasSegmentRunning" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)